karpenter-operator/controllers/nodeclass/ec2_nodeclass_controller.go

```diff
@@ -155,17 +159,21 @@ func (r *EC2NodeClassReconciler) Reconcile(ctx context.Context, req ctrl.Request
 		return ctrl.Result{}, err
 	}
 
-	if _, err := r.CreateOrUpdate(ctx, r.guestClient, ec2NodeClass, func() error {
-		return reconcileEC2NodeClass(ec2NodeClass, openshiftEC2NodeClass, hcp, userDataSecret)
-	}); err != nil {
-		return ctrl.Result{}, err
+	// Don't manage the EC2NodeClass during an upstream karpenter e2e test.
+	karpenterCoreE2E := openshiftEC2NodeClass.Annotations[KarpenterCoreE2EOverrideAnnotation] == "true"
```

Comment:
I don't think we want to allow users to set this from the guest cluster. I think this should be controller from the management side, i.e. the annotation should be on the HCP instead.

Reply:
Fixed in https://github.com/openshift/hypershift/pull/5680/commits/f79c29da10fea9f027e36a57b6026d7252456348